### DISCUSSION TITLE
Testing dataset

### DISCUSSION THREAD

**Discussion by cece3014** [talmolab/sleap#865] [Help!]
Hi All, 

First, wow! SLEAP is absolutely incredible. As I am relatively new to computer vision and machine learning, the interface is very easy to follow so thank you! 
I am currently running training on my model, but had a question regarding testing. In my understanding of deep learning, the final step is to test the trained model on unseen video samples and evaluate accuracy. Perhaps I've missed it somewhere, but is this possible to do on SLEAP after we've proofread our labels on a trained model? 

Thanks for all your time!

**Comment by talmo** [comment]
Hi @cece3014,

In SLEAP we do this automatically when you train -- you set a validation fraction (the fraction of all labels that will be held out) and it gets randomly sampled before training begins. The `training_config.json` file in the model directory contains the indices to the frames that were used for training vs validation if you need to look that back up, but the metrics are automatically computed on both the training and validation splits and stored in the model directory as well.

Check out [this notebook](https://sleap.ai/notebooks/Model_evaluation.html) for a bit more on evaluation, including on how to do it from scratch given a trained model and some new (or held out) test data.

Cheers,

Talmo
